models/src/curve_vault.rs

```diff
@@ -0,0 +1,239 @@
+            .bind(self.curve_number.to_big_decimal()?)
+            .bind(self.total_shares.to_big_decimal()?)
+            .bind(self.current_share_price.to_big_decimal()?)
+            .bind(self.position_count)
+            .fetch_one(pool)
+            .await
+            .map_err(|e| ModelError::InsertError(e.to_string()))
+    }
+
+    /// Finds a curve vault by its composite key (atom_id, triple_id, curve_number).
+    async fn find_by_id(
+        id: (Option<U256Wrapper>, Option<U256Wrapper>, U256Wrapper),
+        pool: &PgPool,
+        schema: &str,
+    ) -> Result<Option<Self>, ModelError> {
+        let (atom_id, triple_id, curve_number) = id;
+        
+        let query = format!(
+            r#"
+            SELECT 
+                id,
+                atom_id, 
+                triple_id,
+                curve_number,
+                total_shares, 
+                current_share_price,
+                position_count
+            FROM {}.curve_vault 
+            WHERE 
+                (atom_id = $1 OR (atom_id IS NULL AND $1 IS NULL)) AND
+                (triple_id = $2 OR (triple_id IS NULL AND $2 IS NULL)) AND
+                curve_number = $3
+            "#,
+            schema,
+        );
+
+        sqlx::query_as::<_, CurveVault>(&query)
+            .bind(atom_id.as_ref().and_then(|w| w.to_big_decimal().ok()))
+            .bind(triple_id.as_ref().and_then(|w| w.to_big_decimal().ok()))
+            .bind(curve_number.to_big_decimal()?)
+            .fetch_optional(pool)
+            .await
+            .map_err(|e| ModelError::QueryError(e.to_string()))
+    }
+}
+
+impl CurveVault {
+    /// Updates the current share price of a curve vault.
+    pub async fn update_current_share_price(
+        atom_id: Option<U256Wrapper>,
+        triple_id: Option<U256Wrapper>,
+        curve_number: U256Wrapper,
+        current_share_price: U256Wrapper,
+        pool: &PgPool,
+        schema: &str,
+    ) -> Result<Self, ModelError> {
+        let query = format!(
+            r#"
+            UPDATE {}.curve_vault 
+            SET current_share_price = $1 
+            WHERE 
+                (atom_id = $2 OR (atom_id IS NULL AND $2 IS NULL)) AND
+                (triple_id = $3 OR (triple_id IS NULL AND $3 IS NULL)) AND
+                curve_number = $4
+            RETURNING id, atom_id, triple_id, curve_number, total_shares, current_share_price, position_count
+            "#,
+            schema,
+        );
+
+        sqlx::query_as::<_, CurveVault>(&query)
+            .bind(current_share_price.to_big_decimal()?)
+            .bind(atom_id.as_ref().and_then(|w| w.to_big_decimal().ok()))
+            .bind(triple_id.as_ref().and_then(|w| w.to_big_decimal().ok()))
+            .bind(curve_number.to_big_decimal()?)
+            .fetch_one(pool)
+            .await
+            .map_err(|e| ModelError::UpdateError(e.to_string()))
+    }
+
+    /// Finds curve vaults by atom ID.
+    pub async fn find_by_atom_id(
```

Comment:
Lower priority: We could combine these, as the return type is the same and the atoms and triples are indexed the same way.

But defer to the base vault implementation, as we should just stay consistent with it.